lib/puppet/type/foreman_auth.rb

```diff
@@ -0,0 +1,121 @@
+require 'puppet/parameter/boolean'
+Puppet::Type.newtype(:foreman_auth) do
+  @doc = 'foreman_auth manages LDAP authentication sources in foreman.'
+  ensurable
+
+  newproperty(:account) do
+    desc 'Bind account name for browsing LDAP directory.'
+    defaultto ''
+  end
+
+  newproperty(:account_password) do
+    desc 'Bind account password for browsing LDAP directory.'
+    defaultto ''
+  end
+
+  newproperty(:attr_firstname) do
+    desc ''
+    defaultto 'givenName'
+  end
+
+  newproperty(:attr_lastname) do
+    desc ''
+    defaultto 'sn'
+  end
+
+  newproperty(:attr_login) do
+    desc ''
+    defaultto 'sAMAccountName'
+  end
+
+  newproperty(:attr_mail) do
+    desc ''
+    defaultto 'mail'
+  end
+
+  newproperty(:attr_photo) do
+    desc ''
+    defaultto ''
+  end
+
+  newproperty(:base_dn) do
+    desc 'Top level DN of your LDAP directory tree'
+    defaultto ''
+  end
+
+  newparam(:base_url) do
+    desc 'Foreman\'s base url.'
+  end
+
+  newparam(:consumer_key) do
+    desc 'Foreman oauth consumer_key'
+  end
+
+  newparam(:consumer_secret) do
+    desc 'Foreman oauth consumer_secret'
+  end
+
+  newparam(:effective_user) do
+    desc 'Foreman\'s effective user for the registration (usually admin).'
+    defaultto 'admin'
+  end
+
+  newproperty(:host) do
+    desc 'LDAP server hostname'
+    defaultto ''
+  end
+
+  newproperty(:groups_base) do
+    desc 'LDAP server hostname'
+    defaultto ''
+  end
+
+  newparam(:name, namevar: :true) do
+    desc 'Name of the LDAP authentication source.'
+  end
+
+#  newproperty(:onthefly_register, :boolean => true, :parent => Puppet::Parameter::Boolean) do
```

Comment:
Instead of `Puppet::Parameter::Boolean`, can you use `Puppet::Property::Boolean` here?
eg https://github.com/Katello/puppet-pulp/blob/479f65c1e7531e2ab4f174725d2c58d842cc1339/lib/puppet/type/pulp_rpmrepo.rb#L68